specs/change_password/01_main.rst

```diff
@@ -0,0 +1,175 @@
++--------+-----------------+
+| issue  | title           |
++--------+-----------------+
+| 74_    | Change password |
++--------+-----------------+
+
+.. _74: https://github.com/gnosis/safe/issues/74
+
+Change password
+===============
+
+1. `Main`_
+2. `User Interface`_
+3. `External Communication`_
+4. Other_
+
+.. _Main:
+
+.. contents:: Table of Contents
+    :depth: 3
+
+1. Problem Definition
+---------------------
+
+* User wants to change the current app password.
+* Current password was compromised and needs to be changed.
+
+2. Inputs
+-----------
+
+- Current password.
+    - Minimum 8 characters (>=8)
+    - At least 1 number
+    - At least 1 letter
+    - No more than 2 identical characters in a row
+      (e.g. 111 is not allowed, 11 is fine)
+    - Maximum lenght?
+- New password
+    - Same requirements as above.
+- Repeat new password
+    - Needs to match
+
+3. Outputs
+------------
+
+- Changed password
+- Error messages
+  - Wrong current password entered
+  - New password does not meet requirements
+  - Repeated new password does not match.
+
+
+4. Use Case Scenarios
+-----------------------
+
+Every use case scenario consists of preconditions – prerequisites
+and assumptions, steps - numbered sequence of steps of the scenario,
+from the user's perspective, and postconditions - results and
+assumptions that must hold after the steps finished.
+
+A use case can *inherit* another use case. That means that
+inheriting use case takes all of the preconditions, steps, and postconditions
+from the inherited use case.
+
+The inheriting (child) use case
+can override (replace) any step or precondition in the parent
+use case. The overriden steps in child should match the step number
+in parent. The child use case may introduce sub-steps, in that case
+numbering is starts with dot, for example "3.1", "3.2" gives more
+details about step "3" of a parent use case.
+
+
+4.1. Happy Case
+~~~~~~~~~~~~~~~
+
+Preconditions
++++++++++++++
+
+- App is setup with a password
+- User knows the password.
+
+Steps
++++++
+
+1. Open app Settings
+2. Tap "Change password"
+3. Enter the current password
+4. Enter valid new password
+5. Repeat new password
+6. Save
+
+Postconditions
+++++++++++++++
+
+- User is on the settings screen.
+- App password is changed.
+- User can unlock the app with the new password.
+- User cannot unlock the app with the another password.
+
+
+4.2. Happy Case with fingerprint
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+inherits from `4.1. Happy Case`_
+
+Preconditions
++++++++++++++
+
+- App is setup with a password
+- App is setup with fingerprint
+- User knows the password.
```

Comment:
we can omit the repeating postconditions because they are specified in happy case's postconditions, and duplicate information is harder to change in several places.

Reply:
But there is 1 more pre condition: "App ios setup with fingerprint" That's why I repeated them.